transformer_engine/pytorch/attention.py

```diff
@@ -3207,6 +3207,7 @@ def __init__(
         cp_group: Optional[dist_group_type] = None,
         cp_global_ranks: List[int] = None,
         cp_stream: torch.cuda.Stream = None,
+        kv_channels: int = None # deprecated
```

Comment:
Well, if it is deprecated, then we should make sure that the code that explicitly calls `kv_channels=something` still works as before, right? While those changes are just ignoring it.

Reply:
I think the code will crash when kv_channels were set for different value than default, so we can safely do this. We have in the code 
```
self.hidden_size_kv = int(hidden_size * self.num_gqa_groups // num_attention_heads)
...
self.layernorm_qkv = LayerNormLinear(
                    hidden_size,
                    hidden_size + 2 * self.hidden_size_kv,
                    ...
```

So the input size to the attention does not depend from kv_channels - do kv_channels can have one value.

I see this is not the case with DotProductAttention, I will look at this. 

Nevertheless, I want to ask if I want to deprecate an argument, what should I do? Will causing it to behave the same at previously, but removing from docs will be ok? Do we consider it as breaking the API?